src/handlers/annotate.ts

```diff
@@ -122,13 +187,57 @@ async function handleSimilarIssuesComment(
   });
 }
 
+// Process similar comments and return the list of similar comments with their similarity scores
+async function processSimilarComments(
+  similarComments: CommentSimilaritySearchResult[],
+  context: Context,
+  commentBody: string
+): Promise<CommentGraphqlResponse[]> {
+  const processedComments = await Promise.all(
+    similarComments.map(async (comment: CommentSimilaritySearchResult) => {
+      try {
+        const commentUrl: CommentGraphqlResponse = await context.octokit.graphql(
+          /* GraphQL */
+          `
+            query ($commentNodeId: ID!) {
+              node(id: $commentNodeId) {
+                ... on IssueComment {
+                  body
+                  url
+                  issue {
+                    number
+                    repository {
+                      name
+                      owner {
+                        login
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          { commentNodeId: comment.comment_id }
+        );
+        commentUrl.similarity = Math.round(comment.similarity * 100).toString();
+        commentUrl.mostSimilarSentence = findMostSimilarSentence(commentBody, commentUrl.node.body, context);
+        return commentUrl;
+      } catch (error) {
+        context.logger.error(`Failed to fetch comment ${comment.comment_id}: ${error}`, { comment });
+        return null;
+      }
+    })
+  );
+  return processedComments.filter((comment): comment is CommentGraphqlResponse => comment !== null);
+}
+
 /**
  * Checks if a annotate footnote exists in the content.
  * @param content The content to check for annotate footnotes
  * @returns True if a annotate footnote exists, false otherwise
  */
 export function checkIfAnnotateFootNoteExists(content: string): boolean {
-  const footnoteDefRegex = /\[\^(\d+)\^\]: \d+% similar to issue: [^\n]+(\n|$)/g;
+  const footnoteDefRegex = /\[\^(\d+)\^\]: \d+% similar to (issue|comment): [^\n]+(\n|$)/g;
```

Comment:
I still think that the regex is so shaky. If its based on the output of another plugin, this will lead to hard to diagnose bugs in the future.